src/clients/VestingData.ts

```diff
@@ -26,6 +37,42 @@ function getVestingDates(contractStart: number, contractEndsTimestamp: number) {
   }
 }
 
+async function getVestingContractLogs(vestingAddress: string, web3: Web3, version: ContractVersion) {
+  const eth = web3.eth
+  const web3Logs = await eth.getPastLogs({
+    address: vestingAddress,
+    fromBlock: 0,
+    toBlock: 'latest',
+  })
+
+  const blocks = await Promise.all(web3Logs.map((log) => eth.getBlock(log.blockNumber)))
+  const logs: VestingLog[] = []
+  const topics = TopicsByVersion[version]
+
+  const getLog = (timestamp: number, topic: string): VestingLog => ({
+    topic,
+    timestamp: toISOString(timestamp),
+  })
+
+  for (const idx in web3Logs) {
+    switch (web3Logs[idx].topics[0]) {
```

Comment:
we can have a `const` for `Number(blocks[idx].timestamp)` (transactionTimestamp?)